src/Components/ProductCard.jsx

```diff
@@ -1,8 +1,11 @@
 import React from 'react';
+import { Link } from 'react-router-dom';
 import { useCart } from '../Context/cart-context';
+import { useWishlist } from '../Context/wishlist-context';
 
-function ProductCard({ product }) {
+function ProductCard({ product, page }) {
 	const { state, dispatch } = useCart();
+	const { wishlistDispatch } = useWishlist();
 	const checkInCart = () => {
 		for (let i = 0; i < state.cartProducts.length; i++) {
```

Comment:
you could've used higher order function here.

Reply:
Yes, Changed it!